server/app/services/PageNumberBasedPaginationSpec.java

```diff
@@ -0,0 +1,35 @@
+package services;
+
+import java.util.Optional;
+
+/**
+ * Specifies pagination behavior for a query with page number-based pagination. With page-number
+ * based pagination, the current page of results is resolved by multiplying the page size by the
```

Comment:
Suggestion: provide this calculation here, instead of listing the recipe for all users to then apply.   Is there any reason not to?

Reply:
done